nullaway/src/main/java/com/uber/nullaway/NullabilityUtil.java

```diff
@@ -339,17 +322,76 @@ public static boolean nullnessToBool(Nullness nullness) {
     }
   }
 
+  /**
+   * Checks if a symbol comes from an annotated package, as determined by either configuration flags
+   * (e.g. {@code -XepOpt:NullAway::AnnotatedPackages}) or package level annotations (e.g. {@code
+   * org.jspecify.nullness.NullMarked}).
+   *
+   * @param outermostClassSymbol symbol for class (must be an outermost class)
+   * @param config NullAway config
+   * @return true if the class is from a package that should be treated as properly annotated
+   *     according to our convention (every possibly null parameter / return / field
+   *     annotated @Nullable), false otherwise
+   */
+  public static boolean fromAnnotatedPackage(
+      Symbol.ClassSymbol outermostClassSymbol, Config config) {
+    final String className = outermostClassSymbol.getQualifiedName().toString();
+    if (!config.fromExplicitlyAnnotatedPackage(className)
+        && outermostClassSymbol.packge().getAnnotation(NULLMARKED) == null) {
+      // By default, unknown code is unannotated unless @NullMarked or configured as annotated by
+      // package name
+      return false;
+    }
+    if (config.fromExplicitlyUnannotatedPackage(className)) {
+      // Any code explicitly marked as unannotated in our configuration is unannotated, no matter
+      // what.
+      return false;
+    }
+    if (config.shouldTreatGeneratedAsUnannotated()
+        && ASTHelpers.hasDirectAnnotationWithSimpleName(outermostClassSymbol, "Generated")) {
+      // Generated code is or isn't excluded, depending on configuration
+      // Note: In the future, we might want finer grain controls to distinguish code that is
+      // generated with nullability
+      // info and without.
+      return false;
+    }
+    // Finally, if we are here, the code was marked as annotated (either by configuration or
+    // @NullMarked) and
+    // nothing overrides it.
+    return true;
+  }
+
+  /**
+   * Check whether a class or any of its enclosing classes are explicitly marked {@code @NullMarked}
+   *
+   * <p>Important: This method ignores package-level annotations and annotated packages and classes
+   * passed as configuration options. For proper resolution of whether code should be considered
+   * null-annotated, use {@link #isUnannotated(Symbol, Config, NullMarkedCache)} instead.
+   *
+   * @param classSymbol The class to be checked
+   * @return Whether this class or any of its enclosing classes are explicitly marked @NullMarked
+   */
+  public static boolean isClassOrEnclosingAnnotatedNullMarked(
```

Comment:
One thought here is that a couple of these static methods taking `NullMarkedCache` as a parameter (like this one) could instead become instance methods of `NullMarkedCache`; that might improve readability.  But I'm not sure.

Reply:
For now I am going to punt on this refactoring, but maybe add a couple of TODO comments so we don't forgot completely and maybe can address when the spec is more finalized

Reply:
I'm not even sure what to write in the TODO comments 🙂 Will just leave this alone for now